src/Microsoft.Framework.Runtime.Interfaces/IApplicationEnvironment.cs

```diff
@@ -11,6 +12,8 @@ public interface IApplicationEnvironment
         string Version { get; }
         string ApplicationBasePath { get; }
         string Configuration { get; }
+
+        [Obsolete("This method will be removed in a future version. Use IRuntimeEnvironment.Framework instead")]
```

Comment:
What are the rules about deprecating APIs that have not officially shipped (i.e. why not just remove this property)?